### ISSUE TITLE
Feature request: code completion in the repl

### ISSUE THREAD

**Issue by alza-bitz** [jasongilman/proto-repl#148]
Unless I've not set something up right, code completion is not working for me at the repl prompt (for namespaces, function names, vars, and local bindings etc).

It would be great to have this working with proto-repl (for example it's possible with `lein repl`).

N.b. note that completion is working fine for me in source files, it's just not working at the repl prompt.

**Comment by carocad** [comment]
There is already a PR on atom-ink to support this feature on the console. Once it lands, proto-repl would support it automatically on its ink-console.

https://github.com/JunoLab/atom-ink/pull/68

**Comment by jasongilman** [comment]
I agree this would be a good feature. It actually worked with the previous REPL that was a text editor. I'll see how hard it would be to enable in the console. In the meantime you could try out the old one which is still available via the settings.

**Comment by pfitzseb** [comment]
Just to save you some time looking through the undocumented autocomplete-plus API version 3.0 (I'm just around because this issue was mentioned [here](https://github.com/JunoLab/atom-ink/pull/68) for some reason): It should be sufficient to use `scopeSelector` and `textEditorSelector` in your completions provider, e.g. like [this](https://github.com/JunoLab/atom-julia-client/blob/a7582ed51b45fba76347f6d46509226eecdb4af1/lib/runtime/completions.coffee#L13-L14) (plus change the provided version in `package.json`, of course). With the latest ink version everything should just work then.

**Comment by jasongilman** [comment]
@pfitzseb, Thanks! That worked just like you said after a few more changes.

**Comment by ciacci1234** [comment]
Hi @jasongilman I'm running into this issue :sweat: I've spent many hours investigating but no luck, so I'm hoping you might be able to help me out. I'm only trying to get this working for your `proto-repl-demo` project. I have uninstalled Atom and tried with minimal amount of packages necessary per your Atom setup suggestions but still no luck. 

The behavior I am getting is what was described in this original post. Autocomplete in source files works but not in the repl itself. When I disable the Ink Console, the plain text-editor REPL has the expected autocomplete functionality.

The only "errors" showing up in dev console are warnings which I do not believe to be important:

```
IntersectionObserver.observe(target): target element is not a descendant of root.
TreeView @ <embedded>:119293
loadExtensions @ /Users/temp/.atom/packages/emmet/lib/emmet.coffee:156
 Encountered an invalid key binding when adding key bindings from '/Users/temp/.atom/keymap.cson' 'pane:split-right'
```

I looked at the commit which addressed this issue (`e0aab4d`) and all changes that were made within the commit are in place in my version of the project.

I looked at Issue #90 and I have the relevant string you mentioned in that thread. `....m2/repository/clojure-complete/clojure-complete/0.2.4/clojure-complete-0.2.4.jar`

Here is a gist of my current Atom configs:
https://gist.github.com/ciacci1234/ad33085e25c06a8630c4f61321c0197b

Finally, I'm running OS 10.10.5 (Yosemite) and Atom 1.22.1

Other than this issue, I love the project! Thank you for maintaining it :yellow_heart:

**Comment by pfitzseb** [comment]
There was another change in the autocomplete-plus API (followed in a change by the ink API), so you'll need to use something like https://github.com/JunoLab/atom-julia-client/commit/874c4af7c82f633a20725abb15cd61bed7babd23 here as well.

**Comment by ciacci1234** [comment]
Thank you @pfitzseb ✌️  I spent a couple of hours looking into incorporating similar changes into the proto-repl project, but I was unfortunately unsuccessful. I am still fairly new to proto-repl, so I struggled with understanding where to implement these changes in corresponding files for proto-repl

Hopefully a kind reader might be able to point me in the right direction? For the time being, I'll stick with the regular repl-text-editor. 

But thanks again for your quick response about this 👍
